tensorboard/data/server/types.rs

```diff
@@ -89,13 +92,36 @@ impl Borrow<str> for Run {
     }
 }
 
+/// A map defining how many samples per plugin to keep.
+#[derive(Debug)]
+pub struct PluginSamplingHint(pub HashMap<String, usize>);
+
+impl FromStr for PluginSamplingHint {
+    type Err = <usize as FromStr>::Err;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mut result = HashMap::new();
+        for pair_str in s.split(',') {
+            let pair: Vec<_> = pair_str.split('=').collect();
+            if pair.len() != 2 {
+                error!("While parsing plugin sampling hint: {}", s);
+                std::process::exit(1);
```

Comment:
`FromStr` should be a pure function and should not exit or panic. It
should return an `Err` if parsing fails. Feel free to define your own
error type; something like:

```rust
#[derive(Debug, thiserror::Error)]
enum ParsePluginSamplingHintError {
    #[error("component should have exactly 1 \"=\" sign, but found {part:?}")]
    SyntaxError { part: String },
    #[error(transparent)]
    ParseIntError(#[from] std::num::ParseIntError),
}

impl FromStr for PluginSamplingHint {
    type Err = ParsePluginSamplingHintError;
    // ...
}
```